pkg/account/deployer/deployer.go

```diff
@@ -320,6 +358,15 @@ func (d *AccountDeployer) deployServiceUserBindings(ctx context.Context, service
 	}
 }
 
+func (d *AccountDeployer) upsertBoundary(ctx context.Context, boundary account.Boundary) (remoteId, error) {
+	data := accountmanagement.PolicyBoundaryDto{
+		Name:          boundary.Name,
+		BoundaryQuery: boundary.Query,
+	}
+
+	return d.accClient.upsertBoundary(ctx, boundary.OriginObjectID, data)
```

Comment:
At this point, I'm wondering if it makes sense add something like this
```suggestion
oid := boundary.OriginObjectID
foundId := d.idMap.getBoundaryUUID(boundary.ID)
if foundId != "" {
  oid = foundId
}
return d.accClient.upsertBoundary(ctx, oid, data)
```

Reply:
Not only for boundaries though but for all resources

Reply:
It can be a follow-up, though. I think we can reduce the amount of API calls here